src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -336,6 +333,16 @@ static void CheckForExperimentStatus(ITextView textView, Document document)
                 }
             }
 
+            // We need to remember the trigger location for when a completion service claims expanded items are available
+            // since the intial trigger we are able to get from IAsyncCompletionSession might not be the same (e.g. in projection scenarios,)
+            // so when they are requested via expander later, we can retrieve it,.
+            // Techinically we should save the trigger location for each individual service that made such claim, but in reality , only Roslyn's completion service uses
+            // expander, so we can get away with not making such distinction.
```

Comment:
i really don't get teh explanation.

Reply:
i think this might actually be an issue with how Editor handle expander in old razor editor. 

In the example below, when completion is triggered, our implementation of `GetCompletionContextAsync` is called twice, once with trigger location in C# buffer, another with trigger location in TS buffer. However, when user then hit expander, `GetExpandedCompletionContextAsync` is only called once with triggerLocation pointing to the HTML buffer, as a result we don't know which underlying document we should ask for expanded item for (i.e. cs vs ts). 

![image](https://user-images.githubusercontent.com/788783/127071672-2e58d7ca-d390-44bc-b68e-c2d33ce12372.png)

But as mentioned in the comment, only Roslyn's provider actually use Expander, so in reality this will not cause any problem.